### ISSUE TITLE
1192.2.0 alpha can't update to 1221.0.0

### ISSUE THREAD

**Issue by jtower525** [coreos/bugs#1656]
# Issue Report #

### CoreOS Version ###

```
core@bdb-node-05 ~ $ cat /etc/os-release 
NAME=CoreOS
ID=coreos
VERSION=1192.2.0
VERSION_ID=1192.2.0
BUILD_ID=2016-10-21-0026
PRETTY_NAME="CoreOS 1192.2.0 (MoreOS)"
```

### Environment ###

What hardware/cloud provider/hypervisor is being used to run CoreOS?

### Expected Behavior ###

should update OS to latest version

### Actual Behavior ###

remains at current version following a reboot

### Reproduction Steps ###

run 'update_engine_client -update'
check is made, download runs, system reboots (or asks for a reboot)
boots into same version

you can do this multiple times with no change

### Other Information ###

core@bdb-node-05 ~ $ cat /etc/coreos/update.conf 
GROUP=alpha
REBOOT_STRATEGY=etcd-lock

logs are too big to paste here, i'll put in a file and update

### Environment ###

What hardware/cloud provider/hypervisor is being used to run CoreOS?

**Comment by crawford** [comment]
I have an old DigitalOcean droplet that is exhibiting this behavior as well. My newer nodes are fine though. Which provider are you using? How old is this machine?

**Comment by jtower525** [comment]
it's internally hosted on vmware esx.  been running for about a year, has done many OS updates during that time.

**Comment by jtower525** [comment]
and another data point, my other cluster nodes updated from 1192.3.0 to 1221.0.0 just fine, it's the 1192.2.0 hosts that fail.

**Comment by crawford** [comment]
1192.3.0 was never released. Do you mean 1185.3.0?

**Comment by crawford** [comment]
For reference, my machine was originally booted as 522.1.0.

**Comment by crawford** [comment]
I successfully updated from 522.1.0 to 1221.0.0 on AWS.

**Comment by crawford** [comment]
@jtower525:

```
sudo mount -o ro $(sudo cgpt find -t coreos-usr | grep --invert-match "$(findmnt --noheadings --raw --output=source --target=/usr)") /mnt
cat /mnt/lib/os-release
sudo umount /mnt
```

Do all of your machines have trouble or is it a subset?

**Comment by jtower525** [comment]
[update-engine-logs-2.txt](https://github.com/coreos/bugs/files/584503/update-engine-logs-2.txt)

**Comment by crawford** [comment]
While trying to repro this, I noticed that manual rollbacks don't seem to work.

```
$ findmnt /usr/
TARGET SOURCE     FSTYPE OPTIONS
/usr   /dev/xvda4 ext4   ro,relatime,seclabel,block_validity,delalloc,barrier,user_xattr,acl
```

```
$ sudo cgpt show /dev/xvda
       start        size    part  contents
           0           1          Hybrid MBR
           1           1          Pri GPT header
           2          32          Pri GPT table
        4096      262144       1  Label: "EFI-SYSTEM"
                                  Type: EFI System Partition
                                  UUID: 24B83D4E-7546-47A0-9C00-4F5B47347C28
                                  Attr: Legacy BIOS Bootable
      266240        4096       2  Label: "BIOS-BOOT"
                                  Type: BIOS Boot Partition
                                  UUID: 22D9ED73-4BBB-489E-8169-8E84657BBAC5
      270336     2097152       3  Label: "USR-A"
                                  Type: Alias for coreos-rootfs
                                  UUID: 7130C94A-213A-4E5A-8E26-6CCE9662F132
                                  Attr: priority=1 tries=0 successful=1
     2367488     2097152       4  Label: "USR-B"
                                  Type: Alias for coreos-rootfs
                                  UUID: E03DD35C-7C2D-4A47-B3FE-27F15780A57C
                                  Attr: priority=2 tries=0 successful=1
     4464640      262144       6  Label: "OEM"
                                  Type: Alias for linux-data
                                  UUID: A94033B8-55D6-4E46-9B9B-9C3BC5D7C7B3
     4726784      131072       7  Label: "OEM-CONFIG"
                                  Type: CoreOS reserved
                                  UUID: D399A4C8-66A7-472E-8E67-3E0520F31872
     4857856    11919327       9  Label: "ROOT"
                                  Type: CoreOS auto-resize
                                  UUID: 86396151-61C8-4DDF-B181-74B3EB8E9053
    16777183          32          Sec GPT table
    16777215           1          Sec GPT header
```

```
$ sudo cgpt prioritize "$(sudo cgpt find -t coreos-usr | grep --invert-match "$(findmnt --noheadings --raw --output=source --target=/usr)")"
```

```
$ sudo cgpt show /dev/xvda
       start        size    part  contents
           0           1          Hybrid MBR
           1           1          Pri GPT header
           2          32          Pri GPT table
        4096      262144       1  Label: "EFI-SYSTEM"
                                  Type: EFI System Partition
                                  UUID: 24B83D4E-7546-47A0-9C00-4F5B47347C28
                                  Attr: Legacy BIOS Bootable
      266240        4096       2  Label: "BIOS-BOOT"
                                  Type: BIOS Boot Partition
                                  UUID: 22D9ED73-4BBB-489E-8169-8E84657BBAC5
      270336     2097152       3  Label: "USR-A"
                                  Type: Alias for coreos-rootfs
                                  UUID: 7130C94A-213A-4E5A-8E26-6CCE9662F132
                                  Attr: priority=2 tries=0 successful=1
     2367488     2097152       4  Label: "USR-B"
                                  Type: Alias for coreos-rootfs
                                  UUID: E03DD35C-7C2D-4A47-B3FE-27F15780A57C
                                  Attr: priority=1 tries=0 successful=1
     4464640      262144       6  Label: "OEM"
                                  Type: Alias for linux-data
                                  UUID: A94033B8-55D6-4E46-9B9B-9C3BC5D7C7B3
     4726784      131072       7  Label: "OEM-CONFIG"
                                  Type: CoreOS reserved
                                  UUID: D399A4C8-66A7-472E-8E67-3E0520F31872
     4857856    11919327       9  Label: "ROOT"
                                  Type: CoreOS auto-resize
                                  UUID: 86396151-61C8-4DDF-B181-74B3EB8E9053
    16777183          32          Sec GPT table
    16777215           1          Sec GPT header
```

```
$ sudo reboot
```

```
$ findmnt /usr/
TARGET SOURCE     FSTYPE OPTIONS
/usr   /dev/xvda4 ext4   ro,relatime,seclabel,block_validity,delalloc,barrier,user_xattr,acl
```

```
$ sudo cgpt show /dev/xvda
       start        size    part  contents
           0           1          Hybrid MBR
           1           1          Pri GPT header
           2          32          Pri GPT table
        4096      262144       1  Label: "EFI-SYSTEM"
                                  Type: EFI System Partition
                                  UUID: 24B83D4E-7546-47A0-9C00-4F5B47347C28
                                  Attr: Legacy BIOS Bootable
      266240        4096       2  Label: "BIOS-BOOT"
                                  Type: BIOS Boot Partition
                                  UUID: 22D9ED73-4BBB-489E-8169-8E84657BBAC5
      270336     2097152       3  Label: "USR-A"
                                  Type: Alias for coreos-rootfs
                                  UUID: 7130C94A-213A-4E5A-8E26-6CCE9662F132
                                  Attr: priority=1 tries=0 successful=1
     2367488     2097152       4  Label: "USR-B"
                                  Type: Alias for coreos-rootfs
                                  UUID: E03DD35C-7C2D-4A47-B3FE-27F15780A57C
                                  Attr: priority=2 tries=0 successful=1
     4464640      262144       6  Label: "OEM"
                                  Type: Alias for linux-data
                                  UUID: A94033B8-55D6-4E46-9B9B-9C3BC5D7C7B3
     4726784      131072       7  Label: "OEM-CONFIG"
                                  Type: CoreOS reserved
                                  UUID: D399A4C8-66A7-472E-8E67-3E0520F31872
     4857856    11919327       9  Label: "ROOT"
                                  Type: CoreOS auto-resize
                                  UUID: 86396151-61C8-4DDF-B181-74B3EB8E9053
    16777183          32          Sec GPT table
    16777215           1          Sec GPT header
```

**Comment by crawford** [comment]
Provisioning with 835.13.0 allows me to manually rollback. This strategy should work with older versions of our bootloader. I'm not sure why it doesn't (or if it's even related to the original failure).

**Comment by crawford** [comment]
I was able to update from 522.1.0 -> 1192.2.0 -> 1221.0.0 on AWS. I'll have to try this on another platform to make sure kexec doesn't play into this. I'm at a bit of a loss as to why this is failing.

**Comment by jtower525** [comment]
if there's something in particular that you'd  like me to check on my nodes that fail just say the word.  i also tried a rollback and that failed so i can't go forward or backwards.

**Comment by crawford** [comment]
Do you have console logs available? I'd be curious to see if the machines are attempting to boot 1221.0.0, failing, and rolling back to 1192.2.0 or if they aren't attempting it at all. You could also try forcing the machine to boot from the alternate partition. If you hit a key early in the boot, you can tell the bootloader to boot a specific partition (A or B) vs the default which will pick one automatically.

**Comment by jtower525** [comment]
good call, it's a failed boot on one of the partitions (kernel panic, screenshot attached).  the strange thing is that my other cluster (which upgraded to 1221.0.0 successfully) is running on the same esx hardware.  i'll check to see if there is some minor difference between the vms, they were created a different times so they may have a slightly different config.
<img width="763" alt="screen shot 2016-11-11 at 02 08 14" src="https://cloud.githubusercontent.com/assets/13770592/20202153/0d897892-a7b5-11e6-9511-46bb96f8e0b1.png">

**Comment by jtower525** [comment]
i learned something else tonight.  bdb-node-05 (which kernel panics on coreos 1221) is on vmware hardware level 8.  my other node (which does not kernel panic on coreos 1221) is on vmware hardware level 7.  they are running side by side on the same physical hardware so that's the only difference, or so i am told by the vmware admins.

**Comment by crawford** [comment]
I finally had a chance to dig into this one. My machine appears to panic when it attempts to kexec the target kernel. As a result, it's never able to update. The interesting thing is that there is no backtrace. Here are the interesting bits from my serial logs:

Running 1192.2.0. Reboot:

```
Loading SYSLINUX configuration...
...
dracut-pre-mount[330]: bootengine: mounting PARTUUID=7130c94a-213a-4e5a-8e26-6cce9662f132 to /run/gptprio
...
kexec: Starting new kernel
Loading SYSLINUX configuration...
```

It is at this point that the kexec appears to have failed and the machine started booting again. The serial logs from the remainder of the second attempt, which succeeds:

```
Loading SYSLINUX configuration...
...
dracut-pre-mount[328]: bootengine: mounting PARTUUID=e03dd35c-7c2d-4a47-b3fe-27f15780a57c to /run/gptprio
...
kexec: Starting new kernel
Linux version 4.7.3-coreos-r1 (jenkins@jenkins-os-executor-1.c.coreos-gce-testing.internal) (gcc version 4.9.3 (Gentoo Hardened 4.9.3 p1.5, pie-0.6.4) ) #1 SMP Fri Oct 21 00:17:36 UTC 2016
Command line: init=/usr/lib/systemd/systemd console=ttyS0,115200n8 console=tty0 ro noswap cros_legacy root=LABEL=ROOT rootflags=subvol=root usr=gptprio: coreos.oem.id=digitalocean usr=PARTUUID=e03dd35c-7c2d-4a47-b3fe-27f15780a57c
...
```

The kernel that fails is 4.8.4 (CoreOS 1214.0.0) and the kernel that succeeds is 4.7.3 (CoreOS 1192.2.0). I took a snapshot of this machine and successfully updated it under QEMU, so the kexec crash is related to DigitalOcean's environment (which I believe is libvirt).

**Comment by euank** [comment]
The 4.8 kernel had hotplug issues which were fixed in 4.9 and via a backport to https://github.com/coreos/linux/pull/40

Because the above stacktrace sreenshot references "rapl_cpu_offline", it seems quite likely that was due to the same kernel bug.

These should be fixed in all channels right now, and more specifically in alpha >= 1325.0.0

Please reopen this bug if my diagnosis is wrong and you still encounter panics/issues on update.
